common/src/androidTest/java/com/microsoft/identity/common/AzureActiveDirectoryAuthorizationResultFactoryTest.java

```diff
@@ -0,0 +1,258 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationResponse;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationResult;
+import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryAuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+
+@RunWith(AndroidJUnit4.class)
+public class AzureActiveDirectoryAuthorizationResultFactoryTest {
+
+    private static final String REDIRECT_URI = "aad-clientid://packagename/";
+    private static final String AUTH_CODE_AND_STATE = "code=authorization_code&state=state";
+    private static final String AUTH_CODE = "authorization_code";
+    private static final String STATE = "state";
+    private static final String ERROR_MESSAGE = "access_denied";
+    private static final String ERROR_DESCRIPTION = "access denied error description";
+    private static final String CORRELATION_ID = "correlationId";
+    private static final String ERROR_CODES = "access_denied_error_code";
+
+    @Test
+    public void testBrowserCodeCancel() {
+        Intent intent = new Intent();
+        intent.putExtras(new Bundle());
+        AzureActiveDirectoryAuthorizationResultFactory factory = new AzureActiveDirectoryAuthorizationResultFactory();
```

Comment:
Once this factory is genericized, you can change the left-side declaration to: `AuthorizationResultFactory<AzureActiveDirectoryAuthorizationResult>` for lower coupling inside of this test.

If you want to _really_ trick this out, you can declare the `AuthorizationResultFactory<AzureActiveDirectoryAuthorizationResult>` as a field in this class and then implement

```java
@Before
public void setUp() {
    mAuthorizationResultFactory = new AzureActiveDirectoryAuthorizationResultFactory();
}
```

That way, if we ever modify this constructor, we only have to update one line of code.

Reply:
Makes sense 👍